src/dmd/hdrgen.d

```diff
@@ -1901,8 +1901,7 @@ public:
         buf.writestring(buffer.ptr);
         if (type)
         {
-            Type t = type.toBasetype();
```

Comment:
Since `floatToBuffer` is only called from the `RealExp` and `ComplexExp` cases, whose type is `TypeBasic`, there's no point in calling `toBasetype`.